responsive-spring/src/test/java/dev/responsive/spring/KafkaStreamsApp.java

```diff
@@ -0,0 +1,151 @@
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.ByteArraySerializer;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Named;
+import org.apache.kafka.streams.kstream.Printed;
+import org.apache.kafka.streams.kstream.TimeWindows;
+import org.apache.kafka.streams.kstream.ValueMapper;
+import org.junit.jupiter.api.Disabled;
+import org.springframework.boot.SpringApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.annotation.EnableKafka;
+import org.springframework.kafka.annotation.KafkaStreamsDefaultConfiguration;
+import org.springframework.kafka.config.KafkaStreamsConfiguration;
+import org.springframework.kafka.config.StreamsBuilderFactoryBeanConfigurer;
+import org.springframework.kafka.config.TopicBuilder;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaAdmin;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+import org.springframework.kafka.listener.DeadLetterPublishingRecoverer;
+import org.springframework.kafka.streams.RecoveringDeserializationExceptionHandler;
+
+@Configuration
+@EnableKafka
+@EnableResponsive
+@Disabled // ignore from tests
+public class KafkaStreamsApp {
+
+  public static void main(String[] args) throws NoSuchMethodException {
+    SpringApplication.run(KafkaStreamsApp.class, args);
+  }
+
+  @Bean(name = KafkaStreamsDefaultConfiguration.DEFAULT_STREAMS_CONFIG_BEAN_NAME)
+  public KafkaStreamsConfiguration streamsConfigs() {
+    Map<String, Object> props = new HashMap<>();
+    props.put(StreamsConfig.APPLICATION_ID_CONFIG, "testStreams");
+    props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
+    props.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+    props.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+    props.put(ResponsiveConfig.STORAGE_BACKEND_TYPE_CONFIG, StorageBackend.MONGO_DB.name());
+    props.put(ResponsiveConfig.MONGO_ENDPOINT_CONFIG, "mongodb://localhost:27017");
+    props.put(StreamsConfig.DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG,
+        RecoveringDeserializationExceptionHandler.class);
+    props.put(
+        RecoveringDeserializationExceptionHandler.KSTREAM_DESERIALIZATION_RECOVERER,
+        recoverer()
+    );
+    return new KafkaStreamsConfiguration(props);
+  }
+
+  @Bean
+  public StreamsBuilderFactoryBeanConfigurer moreMagic() {
```

Comment:
😬

Reply:
Was laughing at the method name but also just noticed the return type as well -- if that isn't the most Java nonsense I've ever seen...

Reply:
haha I thought it was a little cheeky ha!